src/librustc/ty/maps.rs

```diff
@@ -580,10 +598,17 @@ macro_rules! define_maps {
                        key,
                        span);
 
+                profq_msg!(tcx,
+                    ProfileQueriesMsg::QueryBegin(span.clone(),
+                                                  QueryMsg::$name(format!("{:?}", key))));
```

Comment:
Maybe we want to separate the mechanisms for debugging and profiling more -- it would mean losing the ability to answer "which bits of the input took the most time", but it's not 100% clear to me that this is a useful query. It seems useful in the abstract, but I can't remember asking that question very frequently. But that might be because there wasn't an easy way to do so. Certainly it could be useful when you get random perf regressions. I *do* often want "show me all debug logs pertaining to `foo`", but that's more a debugging thing.